client/styles/task-runner.css

```
@@ -0,0 +1,73 @@
+/* This file is manually maintained, it is not generated from SCSS sources */
+
+.task {
+  padding-top: 10px;
+}
+
+.task__panel {
+  padding: 10px 15px 15px 15px;
+}
+
+.task__selector {
+    display: inline-block;
+    height: 42px;
+    margin: 0 0 0 15px;
+    visibility: hidden;
+}
+
+.task__selector--universal {
+  width: 105px;
+}
+
+.task__selector--immediate {
+  width: 144px;
+}
+
+.task__selector--queue-only {
+  width: 146px;
+}
+
+.task__selector:checked + .task__label .task__label-inner,
+.task__selector:hover + .task__label .task__label-inner {
+  border-bottom-color: #43536d;
+}
+
+.task__label {
+    display: inline-block;
+    transform: translateY(-15px);
+}
+
+.task__label--universal {
+  margin-left: -109px;
+  width: 105px;
+}
+
+.task__label--immediate {
+  margin-left: -147px;
+  width: 144px;
+}
+
+.task__label--queue-only {
+  margin-left: -150px;
+  width: 146px;
+}
+
+.task__label-inner {
+  border-bottom: 2px solid transparent;
+  padding-bottom: 15px;
+  transition: border-bottom-color 0.2s;
+}
+
+.task__label:first-child {
+    margin-left: 0;
+}
+
+.task__panel {
+    display: none;
+}
+
+.task__selector--immediate:checked ~ .task__panel--immediate,
+.task__selector--universal:checked ~ .task__panel--universal,
+.task__selector--queue-only:checked ~ .task__panel--queue-only {
+    display: block;
+}
```

Love the technique - and I think you could take it a step further. At the moment you're grouping the tasks into separate sections, which can result in some inconsistent spacing between items at the split. Instead, you could group them back together, shift the modifier to the `.task__item` level, and use an adjusted selector to hide / show individual tasks:

```suggestion
.task__selector--immediate:checked ~ .task__panel .task__item--immediate,
.task__selector--universal:checked ~ .task__panel .task__item--universal,
.task__selector--queue-only:checked ~ .task__panel .task__item--queue-only {
    display: block;
}
```

This would also mean you could reduce duplication in the template, and instead have conditionals around the task buttons.

Of course, keeping the types split into separate sections might be desirable. If this is the case, I'd at least suggest adding a heading to each section to make this separation clearer.

---

Great suggestion, thanks :)